### ISSUE TITLE
Temperature gradient

### ISSUE THREAD

**Issue by mohamadhamdi** [spirit-code/spirit#360]
Applying temperature gradients is crucial to investigate termomagnonic spin transfer torques.

**Comment by GPMueller** [comment]
Please specify more precisely what you would like to achieve. Would it be sufficient to have the following parameters?
- direction vector
- starting temperature (at first point along the direction vector)
- end temperature (at last point along direction vector) or inclination (would not need an end point)

These parameters would be added to `Parameters_LLG` and would need to be used to calculate the temperature distribution at all spins and applied to the force vector accordingly.

**Comment by mohamadhamdi** [comment]
What you mentioned is exactly what is needed.

**Comment by GPMueller** [comment]
Started implementation on with 4b07744f4fa0bb8a185e84b4313c6fb845464edd on `feature-t-gradient`.
Input parameters are:
```Python
llg_temperature_gradient_direction 1 0 0
llg_temperature_gradient_start     3
llg_temperature_gradient_inclination 0.3
```

**Comment by mohamadhamdi** [comment]
Great! Thank you dear Gideon.

**Comment by GPMueller** [comment]
Implemented the API and GUI features with ccdee75d8e1996cc9d102d98ce19ae81875b910e and 9979afab0b900c83fcef2d1e098726e2297b92fb.
Merged into develop branch with 39eff0c9f09d4af34a5222d45857bf4abedeea6e.

Please try out the new features when you have the time. You can now set the temperature gradient from the UI:
![image](https://user-images.githubusercontent.com/20167491/35826454-0cc8080c-0ab9-11e8-8f8c-a03300e21ef3.png)
Note that all temperature is disabled if the checkbox is not checked. Also, any temperature entered into the first field will act as the "starting value" for the gradient.

**Comment by GPMueller** [comment]
Seems to be working fine, so I'm closing this issue.
